### ISSUE TITLE
Special BGM Buddies still play their music even if I turn the option off

### ISSUE THREAD

**Issue by Sean1220** [RunnersRevival/revival_issues#16]
**Describe the bug**
Buddies that play special background music still play the music even though I turned it off in the item select menu.

**To Reproduce**
1. Go to any mode with buddies that plays special BGM.
2. Toggle the option as off in the item selection menu.
3. The music still plays.

**Expected behavior**
The music still plays.

**Screenshots or videos**
-

**Smartphone (please complete the following information):**
 - Device: Galaxy S21 5G
 - OS: Android 12
 - Version: 2.2.0

**Additional context**
I tried it with Opa-Opa and Opapa.

**Comment by F121Live** [comment]
Does it always happen?
Could you send me a clip of this happening?

**Comment by Sean1220** [comment]
Sorry for the late reply.
The special BGM does not play when the option is toggled as off from the start of the game.
However, once the option is set as on and then turning it off still plays the special BGM. This effect is carried throughout the game, so switching buddies does not reset this predicament until turning the game off and back on. Even after using a buddy that does not have a special BGM and then doing a run with special music buddy while it's toggled off still plays their own music.

In short:
OFF from the start -> Does not play the special BGM.
ON -> OFF -> Plays the special BGM no matter the option until the game is turned off and back on. Does not reset when choosing a different buddy.

Link for the video: https://drive.google.com/file/d/1VYHsHYiqkqloBYaYj9eSo-kbHWsg2Tav/view?usp=sharing

**Comment by FairPlay137** [comment]
Addressed for 2.3.0 Public Beta 1.
